dashboard/app/controllers/lessons_controller.rb

```diff
@@ -17,6 +17,10 @@ def show
   # GET /lessons/1/edit
   def edit
     @lesson_data = @lesson.summarize_for_lesson_edit
+    @related_lessons = @lesson.summarize_related_lessons.map do |lesson|
+      lesson[:lessonEditUrl] = edit_lesson_path(id: lesson[:lessonId])
+      lesson
+    end
```

Comment:
I want to generate this url on the server, so that if/when we change lesson urls later, we won't have to remember to to go and find any client-side logic which might assume the current url pattern of /lessons/123/edit . I can't do this in the summarize method because it is only available within the controller, but i still wanted to keep as much of the summarize logic in the model as possible.

Reply:
I will have to backtrack a bit and say that I found a way to move the `edit_lesson_path` call into the summarize method. the trick is to call `include Rails.application.routes.url_helpers` which gives access to these helper methods.